controllers/bsl.go

```diff
@@ -377,14 +378,53 @@ func (r *DPAReconciler) validateProviderPluginAndSecret(bslSpec velerov1.BackupS
 	return nil
 }
 
-func (r *DPAReconciler) ensureBackupLocationHasVeleroOrCloudStorage(dpa *oadpv1alpha1.DataProtectionApplication) error {
-	for _, bsl := range dpa.Spec.BackupLocations {
-		if bsl.CloudStorage == nil && bsl.Velero == nil {
-			return fmt.Errorf("no bucket CloudStorage or velero BackupStorageLocation provided for backupLocations")
-		}
+func (r *DPAReconciler) ensureBackupLocationHasVeleroOrCloudStorage(bsl *oadpv1alpha1.BackupLocation) error {
+	if bsl.CloudStorage == nil && bsl.Velero == nil {
+		return fmt.Errorf("BackupLocation must have velero or bucket configuration")
```

Comment:
Should we add on top of returning error logging error in the function like this? We could have log.error in the caller, but in my opinion it would make sense to add such error or debug to the function which actually returns the error itself, then we don't need to take care of that logging in any future caller of this func ? This would apply to other parts of this PR.

Reply:
ya, I think this can also be part of the same PR where we can add Types for all these error strings
like,
ErrBucketConfigurationMissing = "BackupLocation must have...."

and then, adding log something like this for all functions?

`if bsl.Velero != nil && bsl.Velero.ObjectStorage != nil && bsl.Velero.ObjectStorage.Prefix == "" && dpa.BackupImages() {
		err:= fmt.Errorf("BackupLocation must have velero prefix when backupImages is not set to false")
		
		r.Log.Error(err, "Validation Failed")

		return err
}`

Reply:
Ok worth noting, so it's not lost, but not within current PR.